src/logic/calculate.js

```diff
@@ -0,0 +1,31 @@
+import operate from './operate';
+
+const calculate = (calData, buttonName) => {
+  let { total, next, operation } = calData;
+  let result;
+  const states = { total, next, operation };
+
+  switch (buttonName) {
+    case '+/-':
+      total = -total;
+      next = -next;
+      operation = buttonName;
+      return total;
+    case 'AC':
+      total = 0;
+      return total;
+    case '.':
+      if (next) {
+        result = [total, '.', next].join('');
+        return result;
+      }
+      break;
+    default:
+      result = operate(total, next, operation);
+      return result;
+  }
+  return states;
```

Comment:
Please consider changing the value of `states` instead of the `switch...case` block and only return states at the end of the `switch...case` block. 

This is because, for example, `case '+/-':` will return only `total` instead of an object with both `total` and `next` inverted.